molgenis-semantic-mapper/src/main/java/org/molgenis/data/mapper/service/impl/AlgorithmServiceImpl.java

```diff
@@ -114,14 +115,14 @@ public void autoGenerateAlgorithm(EntityType sourceEntityType, EntityType target
 				Object result = jsMagmaScriptEvaluator.eval(algorithm, entity);
 				derivedValue = convert(result, targetAttribute);
 			}
-			catch (RuntimeException e)
+			catch (NullPointerException npe)
 			{
-				if (e.getMessage() == null)
-				{
-					return algorithmResult.errorMessage(
-							"Applying an algorithm on a null source value caused an exception. Is the target attribute required?");
-				}
-				return algorithmResult.errorMessage(e.getMessage());
+				return algorithmResult.errorMessage(
+						"Applying an algorithm on a null source value caused an exception. Is the target attribute required?");
```

Comment:
Isn't this the sort of thing we'd like to be able to code? I'd generate (and immediately catch) a coded exception.